### ISSUE TITLE
IF: Test failure: trx_finality_status_forked_if_test 

### ISSUE THREAD

**Issue by heifner** [AntelopeIO/leap#2256]
Failure does not appear related to Lin's changes in his PR.

https://github.com/AntelopeIO/leap/actions/runs/7992024465/attempts/1

```
error 2024-02-21T16:18:20.759 nodeos    controller.cpp:3052           apply_block          ] Validation block id does not match producer block id
debug 2024-02-21T16:18:20.759 net-0     net_plugin.cpp:2644           operator()           ] socket_is_open false, state closed, syncing false, connection 3
debug 2024-02-21T16:18:20.759 net-0     net_plugin.cpp:2644           operator()           ] socket_is_open false, state closed, syncing false, connection 4
error 2024-02-21T16:18:20.759 nodeos    controller.cpp:2922           report_block_header_ ] header_extensions: [[2,"4200000001000101acbdd15a010000000460ae423ad15b974a00010000000100021f1254d7b1aa77512c8898797726b0133be199debe81330792b0c238ee6b0f3e010070ae423ad15b974a0001000000010002291dbffc3835e3b34ec68ce7d3cbb7d6922c257b08ce30fd0dfabf8e34467bfc010080ae423ad15b974a0001000000010002cf3b7298be98cdfdae853e7ee121a6dcb27201b8f714ce69bbf23097004f898d010090ae423ad15b974a000100000001000212d9ec3010d49ffc913f6c269f8c7415fee3cda928e28fc63e3dde64a5708c490100"]] != [[2,"42000000010000"]]
debug 2024-02-21T16:18:20.759 net-3     net_plugin.cpp:2478           sync_recv_block      ] ["localhost:9878 - 87e955e" - 6 127.0.0.1:41668] state in sync
debug 2024-02-21T16:18:20.759 net-2     net_plugin.cpp:3102           process_next_block_m ] ["localhost:9880 - 126b02b" - 7 127.0.0.1:48166] canceling wait, already received block 67, id 17a4f56578fb05e5...
debug 2024-02-21T16:18:20.759 net-2     net_plugin.cpp:2470           sync_recv_block      ] ["localhost:9880 - 126b02b" - 7 127.0.0.1:48166] got block 67
error 2024-02-21T16:18:20.759 nodeos    controller.cpp:2929           report_block_header_ ] b  if: {"new_qc_claim":{"block_num":66,"is_strong_qc":true},"new_proposer_policy":{"schema_version":1,"active_time":"2024-02-21T16:18:30.000","proposer_schedule":{"version":1,"producers":[{"producer_name":"defproducera","authority":[0,{"threshold":1,"keys":[{"key":"EOS58B33q9S7oNkgeFfcoW3VJYu4obfDiqn5RHGE2ige6jVjUhymR","weight":1}]}]},{"producer_name":"defproducerb","authority":[0,{"threshold":1,"keys":[{"key":"EOS5CbcTDgbks2ptTxvyCbT9HFbzX7PDHUY2wN4DDnVBhhQr2ZNDE","weight":1}]}]},{"producer_name":"defproducerc","authority":[0,{"threshold":1,"keys":[{"key":"EOS6Tkpf8kcDfa32WA9B4nTcEJ64ZdDMSNioDcaL6rzdMwnpzaWJB","weight":1}]}]},{"producer_name":"defproducerd","authority":[0,{"threshold":1,"keys":[{"key":"EOS52ntDHqA2qj4xVo7KmxdezMRhvvBqpZBuKYJCsgihisxmywpAx","weight":1}]}]}]}}}
debug 2024-02-21T16:18:20.759 net-2     net_plugin.cpp:2478           sync_recv_block      ] ["localhost:9880 - 126b02b" - 7 127.0.0.1:48166] state in sync
error 2024-02-21T16:18:20.759 nodeos    controller.cpp:2935           report_block_header_ ] ab if: {"new_qc_claim":{"block_num":66,"is_strong_qc":true},"new_proposer_policy":null}
error 2024-02-21T16:18:20.759 nodeos    controller.cpp:3076           apply_block          ] e.to_detail_string(): 3030000 block_validate_exception: Block exception
Block ID does not match
    {"producer_block_id":"0000004317a4f56578fb05e578791f4bbc15bf21eda99198c42c5745b381d519","validator_block_id":"000000430da69d96821f91618d977b7b1f82c781f27aeaf756348e762ee96268"}
    nodeos  controller.cpp:3058 apply_block
 
warn  2024-02-21T16:18:20.759 nodeos    controller.cpp:3420           push_block           ] 3030000 block_validate_exception: Block exception
Block ID does not match
    {"producer_block_id":"0000004317a4f56578fb05e578791f4bbc15bf21eda99198c42c5745b381d519","validator_block_id":"000000430da69d96821f91618d977b7b1f82c781f27aeaf756348e762ee96268"}
    nodeos  controller.cpp:3058 apply_block

    {}
    nodeos  controller.cpp:3084 apply_block

error 2024-02-21T16:18:20.759 nodeos    producer_plugin.cpp:707       operator()           ] Exception on block 67: 3030000 block_validate_exception: Block exception
Block ID does not match
    {"producer_block_id":"0000004317a4f56578fb05e578791f4bbc15bf21eda99198c42c5745b381d519","validator_block_id":"000000430da69d96821f91618d977b7b1f82c781f27aeaf756348e762ee96268"}
    nodeos  controller.cpp:3058 apply_block

    {}
    nodeos  controller.cpp:3084 apply_block
rethrow
    {}
    nodeos  controller.cpp:3420 push_block
```

**Comment by Young178** [comment]
Hello everyone, could you please tell me whether this application is related to "Radar" or "CA", or what is it about? Could you please take a moment to answer it?

**Comment by heifner** [comment]
> Hello everyone, could you please tell me whether this application is related to "Radar" or "CA", or what is it about? Could you please take a moment to answer it?

No. Why do you ask?

**Comment by Young178** [comment]
So what is the application of this?

**Comment by heifner** [comment]
Resolved by #2257
